tests/test_separation.py

```diff
@@ -151,20 +154,38 @@ def __unit_test_default_permutation(metric):
 
 def __unit_test_framewise_small_window(metric):
     # Test for invalid win/hop parameter detection
-    ref_sources = np.random.random_sample((4, 100))
-    est_sources = np.random.random_sample((4, 100))
-    # Rest with window larger than source lengths
-    assert np.allclose(metric(ref_sources, est_sources, window=120, hop=20),
-                       mir_eval.separation.bss_eval_sources(ref_sources,
-                                                            est_sources,
-                                                            False),
-                       atol=A_TOL)
-    # Test with hop larger than source length
-    assert np.allclose(metric(ref_sources, est_sources, window=20, hop=120),
-                       mir_eval.separation.bss_eval_sources(ref_sources,
-                                                            est_sources,
-                                                            False),
-                       atol=A_TOL)
+    if metric == mir_eval.separation.bss_eval_sources_framewise:
```

Comment:
You can collapse this test and the one below by doing something like
```Python
if metric == mir_eval.separation.bss_eval_sources_framewise:
    collapse_fn = mir_eval.separation.bss_eval_sources
elif metric == mir_eval.separation.bss_eval_images_framewise:
    collapse_fn = mir_eval.separation.bss_eval_images
else:
    raise ValueError('Unknown metric {}'.format(metric))
```
then below
```Python
assert np.allclose(metric(ref_sources, est_sources,
                          window=20, hop=120),
                   collapse_fn(ref_sources, est_sources, False),
                   atol=ATOL)
```